### PR TITLE
unpickler: guard against malformed py/reduce values

### PR DESCRIPTION
**Pull request by davvid**
The stored value may not be iterable. Restore to an empty list when invalid py/reduce contents are encountered.

### CODE REVIEW

**Comment by davvid** [comment]
This is similar in spirit to the earlier ossfuzz-detected errors.

**Review by Theelx — APPROVED** [review]
Little suggestion for testing parameters but basically fine.

File: tests/jsonpickle_test.py [hunk 1]
```diff
@@ -317,6 +317,13 @@ def test_iterator_with_invalid_data(unpickler):
         assert next(result) == set()
 
 
+def test_reduce_with_invalid_data(unpickler):
+    """Invalid serialized reduce data results in an empty list"""
+    data = {tags.REDUCE: 0}
```
[the remark this answers is not in the document]
Comment (davvid) [reply tests/jsonpickle_test.py]: Good idea. Doing that immediately brought up additional edge cases that weren't being handled. The latest handles this even more robustly.

After this is merged I can try doing the same for the earlier unpickler updates to see if they need additional checks as well.
